app/src/main/java/com/example/taskyy/ui/screens/AgendaScreen.kt

```diff
@@ -156,29 +147,12 @@ fun AgendaScreen(state: AgendaState, onEvent: (AgendaEvent) -> Unit, timeDateSta
     )
 }
 
-@RequiresApi(Build.VERSION_CODES.O)
 @Composable
-fun StartWorkManager() {
-    val context = LocalContext.current
-    LaunchedEffect(key1 = Unit) {
-        val constraints = Constraints.Builder()
-            .setRequiredNetworkType(NetworkType.CONNECTED)
-            .build()
-        val workRequest = PeriodicWorkRequestBuilder<AgendaItemWorker>(
-            repeatInterval = 2,
-            repeatIntervalTimeUnit = TimeUnit.MINUTES
-        ).setBackoffCriteria(
-            backoffPolicy = BackoffPolicy.LINEAR,
-            duration = Duration.ofSeconds(15)
-        ).setConstraints(constraints)
-            .build()
-        val workManager = WorkManager.getInstance(context)
-        workManager.enqueueUniquePeriodicWork(
-            "AgendaItemWorker",
-            ExistingPeriodicWorkPolicy.KEEP,
-            workRequest
-        )
-    }
+fun StartWorkManager(
+    startWorkManager: (AgendaEvent) -> Unit
+) {
+    val context: Context = LocalContext.current
+    startWorkManager(AgendaEvent.StartWorkManager(context))
 }
```

Comment:
This also doesn't really do anything as a separate function. I'd recommend to stick to launching it from the VM's init